### ISSUE TITLE
elasticache module not idempotent: Fails if cluster already exists

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29876]
From @njsaunders on 2016-02-25T15:52:19Z
##### Issue Type:
- Bug Report
##### Component Name:

elasticache
##### Ansible Version:

```
ansible 2.1.0
  config file =
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

Default
##### Environment:

N/A
##### Summary:

The elasticache module correctly creates an elasticache cluster, but running the same playbook again fails it if the cluster already exists.
##### Steps To Reproduce:

```
    - name: Create Elasticache node
      elasticache:
        name: "my-cluster"
        state: present
        engine: memcached
        node_type: cache.t2.small
        num_nodes: 1
        cache_port: 11112
        wait: no
        hard_modify: no
        region: "us-west-2"
        cache_subnet_group: "my-subnet-group"
        security_group_ids: ["sg-1234567"]
```
##### Expected Results:

Running the above repeatedly should succeed with the first execution creating the cluster and subsequent executions seeing that it exists and moving on
##### Actual Results:

Running the above twice fails with the error:

```
TASK [Create Elasticache node] *************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Cache cluster already exists"}
```

Copied from original issue: ansible/ansible-modules-core#3119

**Comment by ansibot** [comment]
From @ansibot on 2016-02-25T15:52:19Z



@jsdalton, @alachaum, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (103/100 (103%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 5517 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext2.iad.github.net ([192.30.252.193]:58920 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1draLA-002Nj1-15
	for greg@biomassiv.es; Mon, 11 Sep 2017 20:50:16 -0500
Date: Mon, 11 Sep 2017 18:49:35 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505180975;
	bh=FCai5ZgsHOCJIWgpZh+lmf/pdAwi5mffBKXQNlgsTXQ=;
	h=From:Reply-To:To:Cc:Subject:List-ID:List-Archive:List-Post:
	 List-Unsubscribe:From;
	b=NAqsoTgfrFqoGciDhFegD1880XU5OkZTRAgpwtiM+5/sdgnXsoloJLNh2iBVKtGzY
	 ZwUHTyUpoFI5tDuPABx26AY2wGXjhu653FKtqxfSgUaiGTvclI9Q2nKCtvpHdKP/xU
	 32JIkO3zKOGGoLLM7aZR9pMxsb5FSDZ2K1DLSIYE=

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by willthames** [comment]
Tested this with latest Ansible, idempotency seems to now be working as expected.
